File: src/main/scala/org/bakkes/game/model/SpriteNameExtender.java
package org.bakkes.game.model;

/**
 * pokemon have depending on the situation big, small, front or back images (not to speak of the shiny ones)
 * this class allows the views to pick the right ones
 */
public class SpriteNameExtender implements IHasSpriteName{

	private IHasSpriteName target;
	private String extension = "";
	public void setHasSpriteName(final IHasSpriteName spriteName){
		target = spriteName;
	}
	public void setExtension(final String to){
		extension = to;
	}
	@Override
	public String getName() {
		return target.getName();
	}

	@Override
	public void setName(final String to) {
		target.setName(to);
	}

	@Override
	public String getSpriteName() {
		return target.getSpriteName() + extension;
	}

	@Override
	public void setSpriteName(final String to) {
		target.setSpriteName(to);
	}

}

File: src/main/scala/org/bakkes/game/model/SpriteNameExtender.scala
package org.bakkes.game.model

/**
 * pokemon have depending on the situation big, small, front or back images (not to speak of the shiny ones)
 * this class allows the views to pick the right ones
 */
class SpriteNameExtender extends ModelNameExtender with IHasSpriteName {
	private var spriteNamed: IHasSpriteName = null
	val spriteNameExtender = new TStringExtender(){
		override def target = spriteNamed.getSpriteName
	}
	def setHasSpriteName(spriteName: IHasSpriteName) {
		spriteNamed = spriteName
		setModel(spriteNamed)
	}

	override def getSpriteName:String = spriteNameExtender.extend
	override def setSpriteName(to: String) = spriteNamed.setSpriteName(to)
}